File: src/Log/Destruct_Log.php
<?php

namespace xltxlm\logger\Log;

include_once __DIR__ . '/dk_get_dt_id.php';

/**
 * 在类注销的时候,写入日志;
 */
trait Destruct_Log
{
    use Destruct_Log\Destruct_Log_implements;

    public function __construct()
    {
        if ($_REQUEST['c']) {
            $this->setAtcion_entrance(strtr($_REQUEST['c'], ['/' => '@']));
        }
        $this->settimestamp_start(microtime(true));
    }

    /**
     * 如果是异常退出，保证记录下数据
     */
    public function __destruct()
    {
        if ($this->haveloged == false) {
            $this->__invoke();
        }
    }

}

File: go/Log/Destruct_Log.go
package Log

type Destruct_Log struct {
    /* 开始记录日志的时间戳 */
    timestamp_start string

    /* 是否已经主动写入日志 */
    haveloged bool

    /* 记录相同进行下,当条日志的时序 */
    posix_log_num array

    /* 程序执行的入口标志 */
    Atcion_entrance string

    /* 各种日志条数的统计存储 */
    log_cout array

}

func NewDestruct_Log(Atcion_entrance string) *Destruct_Log{
    var this = new(Destruct_Log)
    this.SetAtcion_entrance(Atcion_entrance);
    return this
}

func (this *Destruct_Log)GetAtcion_entrance() string{
    return this.Atcion_entrance;
}

func (this *Destruct_Log)SetAtcion_entrance(Atcion_entrance string) *Destruct_Log{
    this.Atcion_entrance = Atcion_entrance;
    return this
}

/**
    析构的时候,尝试写入日志*/
func (this *Destruct_Log)__destruct(){

}

/**
    手动调用写入日志的接口*/
func (this *Destruct_Log)__invoke(){

}

/**
    初始化入口标志和日志开始计时时间*/
func (this *Destruct_Log)__construct(){

}

